corfudb-tools/src/main/java/org/corfudb/browser/CorfuStoreBrowser.java

```diff
@@ -266,6 +275,98 @@ public int dropTable(String namespace, String tablename) {
         return tableSize;
     }
 
+    public int editRecord(String namespace, String tableName,
+        String keyToEdit, String newRecord) {
+        verifyNamespaceAndTablename(namespace, tableName);
+
+        if (keyToEdit == null) {
+            throw new IllegalArgumentException("Please specify the key to " +
+                "edit --keyToEdit");
+        }
+
+        if (newRecord == null) {
+            throw new IllegalArgumentException("Please specify the newRecord " +
+                "--newRecord");
+        }
+
+        log.info("\n======================\n");
+        String fullName = TableRegistry.getFullyQualifiedTableName(namespace,
+            tableName);
+        UUID streamUUID = UUID.nameUUIDFromBytes(fullName.getBytes());
+        CorfuTable<CorfuDynamicKey, CorfuDynamicRecord> table = getTable(
+            namespace, tableName);
+
+        TableName tableNameProto = TableName.newBuilder().setTableName(tableName)
+            .setNamespace(namespace).build();
+
+        CorfuStoreMetadata.KeyValueDefaultRecord keyValueDefaultRecord =
+            runtime.getTableRegistry().getTableNameToKeyValueTypeTable().get(tableNameProto);
+        Any defaultKeyAny = keyValueDefaultRecord.getKey();
+        Any defaultValueAny = keyValueDefaultRecord.getValue();
+
+        DynamicMessage keyMsg = createMsgFromJson(defaultKeyAny, keyToEdit);
+        DynamicMessage newValueMsg = createMsgFromJson(defaultValueAny,
+            newRecord);
+
+        if (keyMsg == null || newValueMsg == null) {
+            return 0;
+        }
+
+        for (CorfuDynamicKey key : table.keySet()) {
+            if (Objects.equals(keyMsg.toString(), key.getKey().toString())) {
```

Comment:
Although this might work in some cases, it is a wrong comparison in general. A `toString` method usually creates a string representation that is a "summary" and not necessary a complete representation of the object. One example of this, a map can have two different keys that have the same `toString` representation. 

Also, why do you need to iterate the keyset if this is already a hashmap? can't you just get a get a protobuf message for the key and do table.contains(key) ?

Reply:
Agreed.  However, it is not possible to compare field values with equals() in DynamicProtobuf.  Let me know if you have a better suggestion.

Looking at the other comment on iterating vs get() on a key.

Reply:
done.  using table.containsKey().